### PR TITLE
Add Support for CS_INVALIDATED events

### PR DESCRIPTION
**Pull request by pmarkowsky**
Add support for logging when code signing has become invalidated for a process.

This adds support to the Recorder to log when codesigning is invalidated as reported by the Endpoint Security Framework's `ES_EVENT_TYPE_NOTIFY_CS_INVALIDATED` event.

### CODE REVIEW

File: Source/santad/Logs/EndpointSecurity/Serializers/BasicString.mm [hunk 1]
```diff
@@ -412,6 +413,17 @@ static inline void AppendUserGroup(std::string &str, const audit_token_t &tok,
...
 
+std::vector<uint8_t> BasicString::SerializeMessage(const EnrichedCSInvalidated &msg) {
+  const es_message_t &esm = msg.es_msg();
+  std::string str = CreateDefaultString();
+
+  str.append("action=CODESIGNING_INVALIDATED");
+  AppendProcess(str, esm.process);
+  AppendUserGroup(str, esm.process->audit_token, msg.instigator().real_user(),
+                  msg.instigator().real_group());
+  return FinalizeString(str);
```
Comment (mlw) [inline Source/santad/Logs/EndpointSecurity/Serializers/BasicString.mm:424]: I think it'd be helpful to print add the flags value, probably as a hex string

File: Source/santad/Logs/EndpointSecurity/Serializers/Empty.mm [hunk 2]
```diff
@@ -65,6 +66,12 @@
   return {};
 }
 
+std::vector<uint8_t> Empty::SerializeMessage(const EnrichedCSInvalidated &msg) {
+  return {};
+}
+
+
```
Comment (mlw) [inline Source/santad/Logs/EndpointSecurity/Serializers/Empty.mm:73]: Super nit: More blank lines.

File: Source/santad/Logs/EndpointSecurity/Serializers/Protobuf.mm [hunk 3]
```diff
@@ -600,6 +601,18 @@ static inline void EncodeCertificateInfo(::pbv1::CertificateInfo *pb_cert_info,
   return FinalizeProto(santa_msg);
 }
 
+std::vector<uint8_t> Protobuf::SerializeMessage(const EnrichedCSInvalidated &msg) {
+  Arena arena;
+  ::pbv1::SantaMessage *santa_msg = CreateDefaultProto(&arena, msg);
+
+  ::pbv1::CodesigningInvalidated *pb_cs_invalidated = santa_msg->mutable_codesigning_invalidated();
+  EncodeProcessInfoLight(pb_cs_invalidated->mutable_instigator(), msg.es_msg().version,
```
Comment (mlw) [inline Source/santad/Logs/EndpointSecurity/Serializers/Protobuf.mm:609]: Just noting: This event is a real good reason to add CS flags to `ProcessInfoLight` (but as a separate PR...)

**Review by mlw — CHANGES_REQUESTED** [review]

File: Source/santad/Logs/EndpointSecurity/Serializers/BasicString.mm (hunk 1 above)
Comment (pmarkowsky) [reply Source/santad/Logs/EndpointSecurity/Serializers/BasicString.mm:424]: Good call, will fix.

**Review by mlw — APPROVED** [review]

**Review by mlw — APPROVED** [review]

**Pull request merged**